build.psm1

```diff
@@ -861,33 +861,35 @@ function New-PSOptions {
     Write-Verbose "Using framework '$Framework'"
 
     if (-not $Runtime) {
-        if ($environment.IsLinux) {
-            $Runtime = "linux-x64"
-        } elseif ($environment.IsMacOS) {
-            if ($PSVersionTable.OS.Contains('ARM64')) {
-                $Runtime = "osx-arm64"
-            }
-            else {
-                $Runtime = "osx-x64"
-            }
-        } else {
-            $RID = dotnet --info | ForEach-Object {
-                if ($_ -match "RID") {
-                    $_ -split "\s+" | Select-Object -Last 1
-                }
+        function Get-FirstMatch($String, $Pattern) {
+            $Selection = @($String | Select-String $Pattern)
+            if (-not $Selection) {
+                throw "Cannot match pattern: '${Pattern}'"
             }
+            $Selection[0].Matches.Groups[1].Value
+        }
 
+        try {
+            $Info = dotnet --info
+            Write-Verbose "dotnet --info:`n${Info}"
+
+            $Architecture = Get-FirstMatch $Info '^\s*Architecture:\s+(\w+)$'
+            $Platform = Get-FirstMatch $Info '^\s*OS Platform:\s+(\w+)$'
             # We plan to release packages targeting win7-x64 and win7-x86 RIDs,
             # which supports all supported windows platforms.
             # So we, will change the RID to win7-<arch>
-            $Runtime = $RID -replace "win\d+", "win7"
-        }
-
-        if (-not $Runtime) {
+            $WindowsPlatform = $Architecture[0] -eq 'x' ? 'win7' : 'win'
+            $Platform = $Platform -replace 'windows', $WindowsPlatform
+            # OSX historically reports as 'darwin'
+            $Platform = $Platform -replace 'darwin', 'osx'
+
+            $Runtime = "${Platform}-${Architecture}".ToLower()
```

Comment:
I think it would be much clearer to make it explicit about each platform when constructing the final runtime, something like 
```pwsh
switch ($Platform) {
    'Windows' {
        ## constructing for windows
    }

    'Linux' {
        ## constructing for Linux
    }

    default {
        ## constructing for macOS
    }
}
```

Reply:
Agreed, done.